tests/js/searchtools.js

```diff
@@ -1,20 +1,21 @@
 describe('Basic html theme search', function() {
 
+  function loadFixture(name) {
+      req = new XMLHttpRequest();
+      req.open("GET", `base/tests/js/fixtures/${name}`, false);
+      req.send(null);
+      return req.responseText;
+  }
+
   describe('terms search', function() {
 
     it('should find "C++" when in index', function() {
-      index = {
-        docnames:["index"],
-        filenames:["index.rst"],
-        terms:{'c++':0},
-        titles:["&lt;no title&gt;"],
-        titleterms:{}
-      }
-      Search.setIndex(index);
+      eval(loadFixture("cpp/searchindex.js"));
```

Comment:
This test case has begun failing, and the reason is that the token `C++` (as found in a source document) is typically reduced down to the single-character term `c` in the search index.

I think it's best to resolve this by using `Search.parseQuery` to form a query token that will match `c`, instead of trying to get a literal term `c++` into the index (which is possible, but perhaps not best practice).